openvpn/service_discovery/dto/service_proposal.go

```diff
@@ -0,0 +1,78 @@
+package dto
+
+import (
+	"github.com/mysterium/node/service_discovery/dto"
+	"encoding/json"
+	"log"
+)
+
+type proposal struct {
+	Id                int              `json:"id"`
+	Format            string           `json:"format"`
+	ServiceType       string           `json:"service_type"`
+	ProviderId        string           `json:"provider_id"`
+	PaymentMethodType string           `json:"payment_method_type"`
+	ServiceDefinition *json.RawMessage `json:"service_definition"`
+	PaymentMethod     *json.RawMessage `json:"payment_method"`
+	ProviderContacts  []dto.Contact    `json:"provider_contacts"`
+}
+
+func BuildServiceProposalFromJson(jsonData []byte) dto.ServiceProposal {
+	proposal := proposal{}
+
+	if err := json.Unmarshal([]byte(jsonData), &proposal); err != nil {
+		log.Fatal(err)
+	}
+
+	ovProposal := dto.ServiceProposal{
+		Id:                proposal.Id,
+		Format:            proposal.Format,
+		ServiceType:       proposal.ServiceType,
+		ProviderId:        dto.Identity(proposal.ProviderId),
+		PaymentMethodType: proposal.PaymentMethodType,
+		ProviderContacts:  proposal.ProviderContacts,
+	}
+
+	switch proposal.ServiceType {
+	case "openvpn":
+		ovProposal.ServiceDefinition = getServiceDefinitionFromJson(proposal.ServiceDefinition)
```

Comment:
I think it's possible to stop depending on Openvpn implementation in this switcher:

DTODiscovery would be type agnostic:
```
type serviceDefinitionUnserializer func(*RawMessage) ServiceDefinition

var serviceDefinitionMap map[string]serviceDefinitionUnserializer

func RegisterServiceDefinition(serviceType string, unserializer serviceDefinitionUnserializer) {
    serviceDefinitionMap[serviceType] = unserializer
}
```

DTOOpenvpn hooks on top:
```
dto_discovery.RegisterServiceDefinition("openvpn", func(rawDefinition *json.RawMessage) ServiceDefinition {
	var definition dto_openvpn.ServiceDefinition
	if err := json.Unmarshal(*rawDefinition, &definition); err != nil {
		log.Fatal(err)
	}

	return definition
}
```